### PR TITLE
automate and document deployment

### PR DESCRIPTION
**Pull request by ryanatkn**
This is a first pass at automating and documenting the deployment. We can now do `npm run build && npm run deploy` and everything efficiently updates.

There are a few caveats:

1. Initial deployment is not yet automated and its documentation has gaps. I'm kicking this can down the road until we have an immediate need for it. Waiting will also let the current deployment strategy settle a bit as some obvious improvements will likely surface.

2. The `package.json` deploy commands have a gross hack that copies `deploy.js` to `deploy.mjs`. We'll be able to remove this hack when we overhaul the build process. I'll work on that soon.

3. The Node process is not resilient to server restarts yet, despite having walked through the setup for it. Something is causing it to error after the OS reboots and pm2's logging isn't capturing the error. It may be as simple as adding a delay through pm2. I'll investigate this later -- server robustness isn't a priority right now.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,6 @@ node_modules/
 *.ignore.*
 
 /static/index.css
-emails.txt
\ No newline at end of file
+emails.txt
```
Comment (greatbacon) [inline .gitignore]: what's this emails.txt file?

File: README.md [hunk 2]
```diff
@@ -15,6 +15,12 @@ and improve our communication.
 - learn [about Felt](https://felt.dev/about)
 - community at [Felt.social](https://felt.social) and [its source code](https://github.com/feltcoop/felt.social)
 
+## docs
+
+- [project](src/project)
```
Comment (greatbacon) [inline README.md]: I like this nested doc structure.

File: src/project/deploy/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
+import dotenv from 'dotenv';
```
Comment (greatbacon) [inline src/project/deploy/deploy.ts]: This is the deployment script I assume?

File: .gitignore (hunk 1 above)
Comment (ryanatkn) [reply .gitignore]: It's the temporary place we're storing the mailing list signups. In this PR I changed the original behavior to write to a better location - to the build directory in development, and to an external data directory in production. Here's where that's done - https://github.com/feltcoop/felt/blob/master/src/server.js#L15 - I'll do this properly in the next PR that includes Postgres.

I deleted this from the `.gitignore` in a commit today after realizing it wasn't needed there any more - https://github.com/feltcoop/felt/commit/46575bbae7754026a6f9e80f93e717ca477c43d6

File: src/project/deploy/deploy.ts (hunk 3 above)
Comment (ryanatkn) [reply src/project/deploy/deploy.ts]: Yep that's it. The header comments for `deploy` (https://github.com/feltcoop/felt/blob/master/src/project/deploy/deploy.ts#L63) and `createDeploymentCommand` (https://github.com/feltcoop/felt/blob/master/src/project/deploy/deploy.ts#L101) spell out what they do in more detail.

File: README.md (hunk 2 above)
Comment (ryanatkn) [reply README.md]: Sweet - I wonder if it makes sense to have backlinks to the parent up the hierarchy?

**Pull request merged**